editors/sc-ide/core/sc_lexer.cpp

```diff
@@ -28,68 +28,68 @@ QVector<ScLexer::LexicalRule> ScLexer::mLexicalRules;
 
     initKeywordsRules();
     initBuiltinsRules();
 
-    mLexicalRules << LexicalRule(Token::RadixFloat, "^\\b\\d+r[0-9a-zA-Z]*(\\.[0-9A-Z]*)?");
+    mLexicalRules << LexicalRule(Token::RadixFloat, "\\b\\d+r[0-9a-zA-Z]*(\\.[0-9A-Z]*)?");
 
     // Never heard of this one? Check the "Literals" help file :)
-    mLexicalRules << LexicalRule(Token::ScaleDegreeFloat, "^\\b\\d+(s+|b+|[sb]\\d+)\\b");
+    mLexicalRules << LexicalRule(Token::ScaleDegreeFloat, "\\b\\d+(s+|b+|[sb]\\d+)\\b");
 
     // do not include leading "-" in Float, as left-to-right algorithm does
     // not know whether it is not rather a binary operator
-    mLexicalRules << LexicalRule(Token::Float, "^\\b((\\d+(\\.\\d+)?([eE][-+]?\\d+)?(pi)?)|pi)\\b");
+    mLexicalRules << LexicalRule(Token::Float, "\\b((\\d+(\\.\\d+)?([eE][-+]?\\d+)?(pi)?)|pi)\\b");
 
-    mLexicalRules << LexicalRule(Token::HexInt, "^\\b0x(\\d|[a-f]|[A-F])+");
+    mLexicalRules << LexicalRule(Token::HexInt, "\\b0x(\\d|[a-f]|[A-F])+");
 
-    mLexicalRules << LexicalRule(Token::SymbolArg, "^\\b[A-Za-z_]\\w*\\:");
+    mLexicalRules << LexicalRule(Token::SymbolArg, "\\b[A-Za-z_]\\w*\\:");
 
-    mLexicalRules << LexicalRule(Token::Name, "^[a-z]\\w*");
+    mLexicalRules << LexicalRule(Token::Name, "[a-z]\\w*");
 
-    mLexicalRules << LexicalRule(Token::Class, "^\\b[A-Z]\\w*");
+    mLexicalRules << LexicalRule(Token::Class, "\\b[A-Z]\\w*");
 
-    mLexicalRules << LexicalRule(Token::Primitive, "^\\b_\\w+");
+    mLexicalRules << LexicalRule(Token::Primitive, "\\b_\\w+");
 
-    mLexicalRules << LexicalRule(Token::Symbol, "^\\\\\\w*");
+    mLexicalRules << LexicalRule(Token::Symbol, "\\\\\\w*");
 
-    mLexicalRules << LexicalRule(Token::Char, "^\\$\\\\?.");
+    mLexicalRules << LexicalRule(Token::Char, "\\$\\\\?.");
 
-    mLexicalRules << LexicalRule(Token::EnvVar, "^~\\w+");
+    mLexicalRules << LexicalRule(Token::EnvVar, "~\\w+");
 
 
-    mLexicalRules << LexicalRule(Token::SingleLineComment, "^//[^\r\n]*");
+    mLexicalRules << LexicalRule(Token::SingleLineComment, "//[^\r\n]*");
 
-    mLexicalRules << LexicalRule(Token::MultiLineCommentStart, "^/\\*");
+    mLexicalRules << LexicalRule(Token::MultiLineCommentStart, "/\\*");
 
-    mLexicalRules << LexicalRule(Token::Operator, "^[\\+-\\*/&\\|\\^%<>=]+");
+    mLexicalRules << LexicalRule(Token::Operator, "[+\\-*/&|\\^%<>=]+");
 }
 
 void ScLexer::initKeywordsRules() {
     QStringList keywords;
+    // clang-format off
     keywords << "arg"
              << "classvar"
              << "const"
              << "super"
              << "this"
              << "var";
+    // clang-format on
 
-    QString keywordPattern = QStringLiteral("^\\b(%1)\\b").arg(keywords.join("|"));
+    QString keywordPattern = QStringLiteral("\\b(%1)\\b").arg(keywords.join("|"));
     mLexicalRules << LexicalRule(Token::Keyword, keywordPattern);
 }
 
 void ScLexer::initBuiltinsRules() {
     QStringList builtins;
+    // clang-format off
```

Comment:
I'd remove these, just let the formatter do its thing.

Reply:
@sonata-chen it looks like clang-format wants to leave the following lines as they were originally: https://github.com/supercollider/supercollider/actions/runs/11573295631/job/32227021915#step:5:133

Reply:
It seems that different version of clang-formats do not have the same behaviour. I'm using a latest one. There is a [issue](https://github.com/llvm/llvm-project/issues/88483) opened on llvm repo.

Reply:
FWIW [we use clang-format 14](https://github.com/supercollider/supercollider/wiki/Cpp-formatting-instructions#clang-format). I don't think we can expect exactly the same behavior from different versions of clang-format - it took some effort to migrate from version 8 to 14...

Reply:
I have set up pre-commit successfully, thanks!